### PR TITLE
Bugfix for Golang template

### LINKED ISSUE #11: rendering template multiple times fails
When html engine has Reload(true), then rendering multiple times, it fails with 'html/template: cannot Parse after Execute'.

```
func Test_HTML_Render_Twice(t *testing.T) {
	engine := New("./views", ".html")
	engine.Reload(true)

	engine.AddFunc("isAdmin", func(user string) bool {
		return user == "admin"
	})
	if err := engine.Load(); err != nil {
		t.Fatalf("load: %v\n", err)
	}

	var buf, buf2 bytes.Buffer
	engine.Render(&buf, "index", map[string]interface{}{
		"Title": "Hello, World!",
	}, "layouts/main")

	err := engine.Render(&buf2, "index", map[string]interface{}{
		"Title": "Hello, World!",
	}, "layouts/main")

	if err != nil {
		fmt.Println(err.Error())
	}

	expect := `<!DOCTYPE html><html><head><title>Main</title></head><body><h2>Header</h2><h1>Hello, World!</h1><h2>Footer</h2></body></html>`
	result := trim(buf2.String())
	if expect != result {
		t.Fatalf("Expected:\n%s\nResult:\n%s\n", expect, result)
	}
}
```

### PR DESCRIPTION
**Pull request by otaku**
Resolve #11 

Attempted to keep it DRY

Test by @toomasj

### CODE REVIEW

**Comment by Fenny** [comment]
Thanks for your PR @otaku, but I tried to test your code but it won't update the template after a file change. Does it work on your end?

File: html/html.go [hunk 1]
```diff
@@ -100,9 +99,7 @@ func (e *Engine) Load() error {
 	e.mutex.Lock()
 	defer e.mutex.Unlock()
 
-	// Set template settings
-	e.Templates.Delims(e.left, e.right)
-	e.Templates.Funcs(e.funcmap)
+	e.Templates = nil
```
Comment (otaku) [inline html/html.go:102]: Clear all templates.  Fixes issue with Parse after Execute

File: html/html.go [hunk 2]
```diff
@@ -137,10 +134,17 @@ func (e *Engine) Load() error {
 		if err != nil {
 			return err
 		}
+		// Allocate template and set template settings
+		if e.Templates == nil {
```
Comment (otaku) [inline html/html.go:138]: Only allocate the template when needed instead of creating an empty template with the name of `e.directory`

**Comment by otaku** [comment]
I only resolved the test case... this now works now and added a valid test.

**Comment by Fenny** [comment]
Thank you @otaku, we went with a different approach. I tagged you in this PR https://github.com/gofiber/template/pull/13

**Pull request closed without merging**